pennylane/devices/qubit/sampling.py

```diff
@@ -349,4 +349,5 @@ def sample_state(
         samples = rng.choice(basis_states, shots, p=probs)
 
     powers_of_two = 1 << np.arange(num_wires, dtype=np.int64)[::-1]
-    return (samples[..., None] & powers_of_two).astype(np.bool8)
+    states_sampled_base_ten = samples[..., None] & powers_of_two
+    return (states_sampled_base_ten > 0).astype(np.int64)
```

Comment:
Why not `np.int8`?

Reply:
updated the PR details a bit, but as you can see in [this CI run](https://github.com/PennyLaneAI/pennylane/actions/runs/6013291622/job/16310476417) from before I switched from int8 to int64, the _old_ device was jitted thinking it was going to get int8, but then it got int64. This is because the conversion from `tape.numeric_type == int <--> jit_type=jnp.int8` happens in one place, which should ideally be device-agnostic.

I _could_ start making device-specific changes/branches in our code, but that would become messy. I could also change the old device API to _also_ use `np.int8`, but I'm worried that some plugins/other devices might have sample-dtype overrides by accident, and then it'll clash with our interface code.

tl;dr it's just not worth the effort rn, we're not low on space

Reply:
That sounds reasonable. I think it's worth making a story for this though. Using 8 times more memory than "necessary" shouldn't be around long term.

Reply:
We are all on board with you there.